easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.15.1-foss-2023a.eb

```diff
@@ -0,0 +1,246 @@
+easyblock = 'PythonBundle'
+
+name = 'TensorFlow'
+version = '2.15.1'
+
+homepage = 'https://www.tensorflow.org/'
+description = "An open-source software library for Machine Intelligence"
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+toolchainopts = {'pic': True}
+
+use_pip = True
+sanity_pip_check = True
+
+
+builddependencies = [
+    ('Bazel', '6.1.0'),
+    ('git', '2.41.0', '-nodocs'),
+    ('pybind11', '2.11.1'),
+    ('UnZip', '6.0'),
+    ('poetry', '1.5.1'),
+    ('Clang', '16.0.6'),
```

Comment:
I'm not sure about using `Clang` here, even if that is what upstream TensorFlow is now doing. In principle the toolchain defines the compiler. I.e. using `foss` means we are using GCC as compiler. We do _have_ Clang toolchains (https://docs.easybuild.io/version-specific/toolchains/) in EasyBuild, but they are not widely used and you'd miss a lot of the dependencies needed for TensorFlow...

Was the only reason to switch to `Clang` that it was a _tested_ build configuration, as per https://www.tensorflow.org/install/source#tested_build_configurations ? I'd say for us that's not per se a reason to use that compiler. But maybe you had other reasons?

Reply:
Yes that was the only reason; thanks for the explanation, I'll work on a GCC build.

Reply:
Yes that was the only reason; thanks for the explanation, I'll work on a GCC build.